### ISSUE TITLE
[BUG] This configuration affects the local creation of the vault and the application.yaml and logging

### ISSUE THREAD

**Issue by matbmoser** [eclipse-tractusx/digital-product-pass#85]
## Introduction
This change was introduced in the following PR #74 which affects drastically the backend functionality in local.
https://github.com/eclipse-tractusx/digital-product-pass/blob/dd30d63a0d6bf2efeba6186a87f6967296d5838a/consumer-backend/productpass/pom.xml#L182-L192

### What is happening

This new configuration in the POM moves the resource directory to another dir and not the the default dir where the application yaml is included.

```bash
  .   ____          _            __ _ _
 /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
 \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
  '  |____| .__|_| |_|_| |_\__, | / / / /
 =========|_|==============|___/=/_/_/_/
 :: Spring Boot ::                (v3.1.0)
2023-07-08T12:18:44.699+02:00  INFO 14104 --- [           main] o.e.tractusx.productpass.Application     : No active profile set, falling back to 1 default profile: "default"
2023-07-08T12:18:45.966+02:00  INFO 14104 --- [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port(s): 8080 (http)
2023-07-08T12:18:45.975+02:00  INFO 14104 --- [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
2023-07-08T12:18:45.976+02:00  INFO 14104 --- [           main] o.apache.catalina.core.StandardEngine    : Starting Servlet engine: [Apache Tomcat/10.1.8]
2023-07-08T12:18:46.118+02:00  INFO 14104 --- [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
2023-07-08T12:18:46.118+02:00  INFO 14104 --- [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 1397 ms
2023-07-08T12:18:46.279+02:00 ERROR 14104 --- [           main] utils.LogUtil                            : |14104|0,02| [EXCEPTION]  [[org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create filepath, Cannot invoke "String.isEmpty()" because "segment" is null] [org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create filepath
2023-07-08T12:18:46.280+02:00 ERROR 14104 --- [           main] utils.LogUtil                            : |14104|0,02| [EXCEPTION]  [[org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create secrets file, [org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create filepath, Cannot invoke "String.isEmpty()" because "segment" is null] [org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create secrets file
2023-07-08T12:18:46.281+02:00  WARN 14104 --- [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'apiController': Unsatisfied dependency expressed through field 'dataService': Error creating bean with name 'dataTransferService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\DataTransferService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'vaultService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\VaultService.class]: Failed to instantiate [org.eclipse.tractusx.productpass.services.VaultService]: Constructor threw exception
2023-07-08T12:18:46.282+02:00  INFO 14104 --- [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
2023-07-08T12:18:46.292+02:00  INFO 14104 --- [           main] .s.b.a.l.ConditionEvaluationReportLogger : 
Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
2023-07-08T12:18:46.312+02:00 ERROR 14104 --- [           main] o.s.boot.SpringApplication               : Application run failed
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'apiController': Unsatisfied dependency expressed through field 'dataService': Error creating bean with name 'dataTransferService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\DataTransferService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'vaultService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\VaultService.class]: Failed to instantiate [org.eclipse.tractusx.productpass.services.VaultService]: Constructor threw exception
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:713) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:693) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:133) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:482) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1416) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:597) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:520) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:973) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:941) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:608) ~[spring-context-6.0.9.jar:6.0.9]
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:733) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:435) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:311) ~[spring-boot-3.1.0.jar:3.1.0]
	at org.eclipse.tractusx.productpass.Application.main(Application.java:74) ~[classes/:na]
Caused by: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'dataTransferService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\DataTransferService.class]: Unsatisfied dependency expressed through constructor parameter 3: Error creating bean with name 'vaultService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\VaultService.class]: Failed to instantiate [org.eclipse.tractusx.productpass.services.VaultService]: Constructor threw exception
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:800) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:245) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1352) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1189) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:560) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:520) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1417) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1337) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.resolveFieldValue(AutowiredAnnotationBeanPostProcessor.java:710) ~[spring-beans-6.0.9.jar:6.0.9]
	... 18 common frames omitted
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'vaultService' defined in file [C:\catenax-ng\tx-digital-product-pass\consumer-backend\productpass\target\classes\org\eclipse\tractusx\productpass\services\VaultService.class]: Failed to instantiate [org.eclipse.tractusx.productpass.services.VaultService]: Constructor threw exception
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:326) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:314) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1352) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1189) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:560) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:520) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:326) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:324) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1417) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1337) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:888) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:791) ~[spring-beans-6.0.9.jar:6.0.9]
	... 31 common frames omitted
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.eclipse.tractusx.productpass.services.VaultService]: Constructor threw exception
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:224) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:110) ~[spring-beans-6.0.9.jar:6.0.9]
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:323) ~[spring-beans-6.0.9.jar:6.0.9]
	... 45 common frames omitted
Caused by: org.eclipse.tractusx.productpass.exceptions.ServiceException: [org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create secrets file, [org.eclipse.tractusx.productpass.services.VaultService] It was not possible to create filepath, Cannot invoke "String.isEmpty()" because "segment" is null
	at org.eclipse.tractusx.productpass.services.VaultService.createLocalVaultFile(VaultService.java:145) ~[classes/:na]
	at org.eclipse.tractusx.productpass.services.VaultService.<init>(VaultService.java:61) ~[classes/:na]
	at java.base/jdk.internal.reflect.DirectConstructorHandleAccessor.newInstance(DirectConstructorHandleAccessor.java:67) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:484) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:211) ~[spring-beans-6.0.9.jar:6.0.9]
	... 47 common frames omitted
```

## How to solve it?
The solution is to add the resource folder at the resources definition in the POM.xml in this way it will take the files and place it in the JAR classes directory.

```xml
        <resources>
            <resource>
                <directory>${project.basedir}/src/main/resources</directory>
            </resource>
            <resource>
                <directory>${project.basedir}/../..</directory>
                <includes>
                    <include>LICENSE</include>
                    <include>NOTICE.md</include>
                    <include>DEPENDENCIES_BACKEND</include>
                </includes>
                <targetPath>META-INF/</targetPath>
            </resource>
        </resources>
```

**Comment by matbmoser** [comment]
I will add this change in the latest PR #84

**Comment by matbmoser** [comment]
The solution was merged and tested in #86
